### DISCUSSION TITLE
To upgrade to version 6.1.0, do you need PHP8.3?

### DISCUSSION THREAD

**Discussion by nloshka404** [firefly-iii/firefly-iii#8268] [Firefly III]
To upgrade to version 6.1.0, do you need PHP8.3?
# composer update
Do not run Composer as root/super user! See https://getcomposer.org/root for details
Continue as root/super user [yes]? yes
Loading composer repositories with package information
Updating dependencies
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - Root composer.json requires php >=8.3 but your php version (8.2.13) does not satisfy that requirement.

**Comment by JC5** [comment]
https://github.com/firefly-iii/firefly-iii/releases/tag/v6.1.0

**Comment by T0xIcB3ast** [comment]
It's a shame there isn't a a _FOSS_ alternative to Firefly III, so that I could ditch this $### -_-

Somehow, the dev thinks it's a good idea to make mandatory a version which was **officially released in November 2023**, while PHP 8.1 and 8.2 still receive security patches.

They forgot that not everyone (especially those who use Softaculous) can upgrade to 8.3, even if they wanted to.
Not even Apple does this (and they are not a good example, to begin with)...

**Reply by BackSeat** [reply]
You can ditch it right now if you want. Or you can run the previous version. Or you can fork this one and backport it. Or you can behave like a spoiled child.

That's the beauty of Open Source: you have the choice.

**Reply by JC5** [reply]
Hey, thanks for giving me the opportunity to respond.

Yes, PHP8.3 has just been released and yes, previous versions still get security updates. A valid point.

I am the only developer. I decide the language level. There is a list of features I've been anticipating in PHP8.3 that I have been wanting to use for a long time. I've decided to move Firefly III to PHP8.3 so I can. I feel confident that I've made the right choice. Your platform provider, and Softaculous will upgrade soon enough I'm sure.

It's important to know I'm responsible for Firefly III and Firefly III *alone*. I don't develop, maintain or dictate what happens over at Softaculous. In fact, they decided to support Firefly III. I am grateful for the extended exposure, but it was their choice. *Your* choice to use them as your Firefly III provider dictates which version is available to you.

I hope this helps. If you find another tool let me know, I'm happy to help you migrate.

**Reply by T0xIcB3ast** [reply]
@BackSeat 

Thanks, but I know when it's time to get rid of the software. I don't think your comment will help me get there faster.
Even though your comment is useful as an _umbrella in a desert_, you are absolutely right that **I do have a choice**.
At this point, I will choose the last option, as it is the easiest one 😉

**Reply by T0xIcB3ast** [reply]
@JC5 
Hello James,

Firstly, I appreciate that you have not taken my critique personally. You do have my admiration for just being one person managing this (for what's worth it), and for taking the responsibility for the actions you choose, even thought I don't agree with most of them!
You do mention that you believe you took the right choice, but I'm still free to disagree with that same choice, but, it's your choice…

Just to give you a context, my frustration is not just because Softaculous is failing to update! In fact, this started a long time ago. 
I've used a lot of their available software and Firefly has always been an issue, starting from being the first one to make me move to PHP 8, even when 7 was still being highly used (even in corporations).
Or maybe it was the installation of the damn importer (another path where I disagree with the decision of having this completely standalone, which is hard to maintain with shared hosting' and their rules).
Or perhaps it's just the lack of good dashboards… I don't know.

I do agree with you on one thing: Softaculous did a good job promoting the software, but from my point of view, they should just remove it from their repository, since it goes a bit against the philosophy of install everything easily with one-click.

Lastly, I greatly appreciate your offer for help in moving to another tool (even though there is a high chance of sarcasm in that sentence), but I'm sure I won't need it. Even if my PHP programming knowledge is lacklustre, I have some years on my back of working with databases daily, so I'm sure I can find a way to export this.


But worry not, I will move as soon as I can, it's nothing urgent, it's just more of a matter of disliking the software as it is and your decisions than anything else. You partially know that (unfortunately) there aren't any alternatives to this software. Even if we include paid ones (which I've used in the past, and gladly paid for them, until some stopped working or other technical issues) the list is very short. **But if you know a good one, feel free to let me know!** 🙂

Either way, I hope you achieve success or whatever is the motive behind your project…
Goodbye.

**Comment by nloshka404** [comment]
Perfectly updated, everything works. The old version of PHP had to be left for other web applications that do not yet know how to work on new versions of PHP.
